whcli/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+	"github.com/taskcluster/webhooktunnel/util"
+	"github.com/taskcluster/webhooktunnel/whclient"
+)
+
+// whclient <clientID> <accessToken> <targetPort>
+const usage = `whcli <clientID> <accessToken> <targetPort>`
+
+func main() {
+	if len(os.Args) != 4 {
+		panic(usage)
+	}
+
+	clientID, accessToken := os.Args[1], os.Args[2]
+	targetPort, err := strconv.Atoi(os.Args[3])
+	if err != nil || targetPort < 0 || targetPort > 65535 {
+		panic(usage)
+	}
+
+	configurer := func() (whclient.Config, error) {
+		creds := &tcclient.Credentials{
+			ClientID:    clientID,
+			AccessToken: accessToken,
+		}
+		myAuth := auth.New(creds)
+		whtResponse, err := myAuth.WebhooktunnelToken()
+		if err != nil {
+			return whclient.Config{}, err
+		}
+		proxyURL := util.MakeWsURL(whtResponse.ProxyURL)
+		return whclient.Config{
+			ID:        whtResponse.TunnelID,
+			Token:     whtResponse.Token,
+			ProxyAddr: proxyURL,
+			UseDomain: true,
+		}, nil
+
+	}
+
+	client, err := whclient.New(configurer)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("URL: %s", client.URL())
```

Comment:
Ideally an option to print to stdout as JSON, close stdout and have no log messages.. if you want to use this from docker-worker, that'll make you happy...